### ISSUE TITLE
Benchmark improvement suggestions

### ISSUE THREAD

**Issue by cpw** [junkdog/artemis-odb#214]
Hi
So, firstly, I'm talking 100% about desktop JVMs, specifically the oracle hotspot JVM, especially 1.8. 

In your benchmarks, there is little visible evidence that any attempt to tune GC has been made, and you haven't documented which versions of the JVM you have tested. Both of these factors are _extremely_ important to a benchmark - the JVM is a very dynamic moving target, and assumptions like "pooling is better" are often made based on an extremely naive spinup performance test. 

I can't talk for dalvik and the other android stuff, there may be significant differences there. As an experienced Java benchmarker I am just disappointed that your benchmarks are not really going to justify the level of bytecode hackery you seem to desire.

Also, Unsafe? _Really_? That's never going to be opened up for reflection and definitely not for API access. Quite honestly, I will confidently assert that data-bundling will not be an effective way of working - the JVM is too setup for OO style structure for you to be able to effectively make this a reality (trust me, I have seen **many** attempts at this for various reasons, and all of them are ineffective because _that's not how the JVM works_).

On the "pooling", you're probably going to see a couple of behaviours - for short term benchmarks, pooling will seem faster, but with a modern JVM, you'll start to notice that the non-pooled is likely to approach the pooled speed _as the JVM continues running_. What happens is that the various generational GC mechanisms have some heuristics that help it to optimize as it runs - remember, JVMs are anticipated to run servers for days weeks and months on end - so short term benchmarks that spin up a 2 minute test are not going to get any chance to tune these kinds of thing. (There's various technical reasons why pooling screws with these heuristics - but in general, the worst idea is to let garbage escape from short term heap ('nursery generation') into long term heap ('tenured generation') - it's a lot more expensive to garbage collect garbage tenured objects than non-tenured objects and pooling is a guarantee of tenured garbage (that's just going to hang around and _get in the way of real garbage_)

In summary, although you have a lot of interesting technology ideas I have skepticism of your premise for some of them. I would be interested in seeing some information on what generated your assumptions.
